News bug fixes

misspastelwitch opened the pull request:
## What does this change?

Resolves issue [#215](https://github.com/fdnd-agency/nebulaxplorer/issues/215)

Fixed issues with importing data from directus to the news page and newscards, and made the number of news pieces counter show the number of individual news pieces taken as data from directus.

## Images

<img width="564" height="728" alt="image" src="https://github.com/user-attachments/assets/11fe5adb-894c-452c-b43a-697f4ea4eec1" />


## How to review

Please run through the code and check for anything that looks wrong in terms of picking up data. 
As an extra, check for bugs :)

misspastelwitch committed:
Fixed directus link on news page

misspastelwitch committed:
small tweaks

added error message for fetch, made "currently available" responsive to number of news pieces

misspastelwitch committed:
removed typo

RenzoAW commented on src/routes/news/+page.server.js:
Great that you pulled the data from Directus.  I do have some remarks on how you can maybe improve this. I don't think it is necessary to have an if statements in the script because the data will always be there. Also, I spoke with @halie404 and we decided we don't need a `.env` file. So `{ DIRECTUS_NEWS }` can just be the Directus link that you need for the data.

misspastelwitch replied:
Hey, I just fixed these in this commit: [9161e73](https://github.com/fdnd-agency/nebulaxplorer/pull/216/commits/9161e73a46a8563d9f452816d781caabb9d54004)

Let me know if I'm missing anything, otherwise we can merge!

RenzoAW reviewed (changes requested):
See the comment below! :)

misspastelwitch committed:
Solved #216

Got rid of if statement in News page and data is now picked up directly instead of via .env file

misspastelwitch committed:
Merge branch 'dev' into news-bug-fixes

RenzoAW reviewed (approved):
This looks good Stella, great work!
